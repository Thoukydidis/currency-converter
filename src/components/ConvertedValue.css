.convertedBox {
  margin: 10px 0 10px 0;
  padding: 5px 10px 5px 10px;
  color: black;
  border: 1px solid;
  border-radius: 5px;
  font-size: small;
  display: flex;
  justify-content: space-between;
}

.convertedInfoContainer {
  display: grid;
}

.compareArrowsIcon {
  transform: rotate(90deg);
}

.historyBox {
  margin: 10px 0 10px 0;
  padding: 5px 10px 5px 10px;
  color: black;
  background-color: beige;
  border-radius: 5px;
  font-size: small;
  display: flex;
  justify-content: space-between;
}

.convertedValue {
  font-size: large;
  font-weight: 400;
}
