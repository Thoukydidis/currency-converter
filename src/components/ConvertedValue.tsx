import React from "react";
import CurrencyList from "currency-list";
import "./ConvertedValue.css";

interface Props {
  currencyFrom: string;
  currencyTo: string;
  convertedValue: string;
  children: JSX.Element;
  isHistory?: boolean;
}

const ConvertedValue = (props: Props) => {
  const { currencyFrom, currencyTo, convertedValue, children, isHistory } =
    props;
  const currencyFromFull = CurrencyList.get(currencyFrom?.toUpperCase())?.name;
  const currencyToFull = CurrencyList.get(currencyTo?.toUpperCase())?.name;

  const equals: string = "equals";
  const amount: string = "1.00";

  if (currencyFrom && currencyTo && convertedValue) {
    return (
      <div className={isHistory ? "historyBox" : "convertedBox"}>
        <div className="convertedInfoContainer">
          <span>
            {amount} {currencyFromFull ? currencyFromFull : currencyFrom}{" "}
            {equals}
          </span>
          <span className="convertedValue">
            {convertedValue} {currencyToFull ? currencyToFull : currencyTo}
          </span>
        </div>
        {children}
      </div>
    );
  } else return null;
};

export default ConvertedValue;
