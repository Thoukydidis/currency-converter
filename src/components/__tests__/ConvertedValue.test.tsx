import React from "react";
import ConvertedValue, { Props } from "../ConvertedValue";
import { cleanup, render } from "@testing-library/react";

describe("test the CurrencyInputConverter component", () => {
  afterEach(cleanup);
  const props: Props = {
    currencyFrom: "EUR",
    currencyTo: "USD",
    convertedValue: "string",
    children: <div />,
  };
  it("should render the CurrencyInputConverter", () => {
    const view = render(<ConvertedValue {...props} />);
    expect(view).toMatchSnapshot();
  });
  it("should render the histoyBox styles", () => {
    const newProps = { ...props, isHistory: true };
    const view = render(<ConvertedValue {...newProps} />);
    expect(view).toMatchSnapshot();
  });

  it("should not show the full name of the currency from and currency to", () => {
    const newProps = { ...props, currencyFrom: "string", currencyTo: "string" };
    const view = render(<ConvertedValue {...newProps} />);
    expect(view).toMatchSnapshot();
  });
});
