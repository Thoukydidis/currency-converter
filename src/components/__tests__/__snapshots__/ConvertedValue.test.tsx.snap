// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the CurrencyInputConverter component should not show the full name of the currency from and currency to 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="convertedBox"
      >
        <div
          class="convertedInfoContainer"
        >
          <span>
            1.00
             
            STRING
             
            equals
          </span>
          <span
            class="convertedValue"
          >
            string
             
            STRING
          </span>
        </div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="convertedBox"
    >
      <div
        class="convertedInfoContainer"
      >
        <span>
          1.00
           
          STRING
           
          equals
        </span>
        <span
          class="convertedValue"
        >
          string
           
          STRING
        </span>
      </div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test the CurrencyInputConverter component should render the CurrencyInputConverter 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="convertedBox"
      >
        <div
          class="convertedInfoContainer"
        >
          <span>
            1.00
             
            Euro
             
            equals
          </span>
          <span
            class="convertedValue"
          >
            string
             
            US Dollar
          </span>
        </div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="convertedBox"
    >
      <div
        class="convertedInfoContainer"
      >
        <span>
          1.00
           
          Euro
           
          equals
        </span>
        <span
          class="convertedValue"
        >
          string
           
          US Dollar
        </span>
      </div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test the CurrencyInputConverter component should render the histoyBox styles 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="historyBox"
      >
        <div
          class="convertedInfoContainer"
        >
          <span>
            1.00
             
            Euro
             
            equals
          </span>
          <span
            class="convertedValue"
          >
            string
             
            US Dollar
          </span>
        </div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="historyBox"
    >
      <div
        class="convertedInfoContainer"
      >
        <span>
          1.00
           
          Euro
           
          equals
        </span>
        <span
          class="convertedValue"
        >
          string
           
          US Dollar
        </span>
      </div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
